File: java/asazu/dev/asazutaiga/lox/TokenType.java
package dev.asazutaiga.lox;

public enum TokenType {
  // １文字トークン
  LEFT_PAREN, // (
  RIGHT_PAREN, // )
  LEFT_BRACE, // {
  RIGHT_BRACE, // }
  COMMA, // ,
  DOT, // .
  MINUS, // -
  PLUS, // +
  SEMICOLON, // ;
  SLASH, // /
  STAR, // *

  // 1 or 2 文字トークン
  BANG, // !
  BANG_EQUAL, // !=
  EQUAL, // =
  EQUAL_EQUAL, // ==
  GREATER, // >
  GREATER_EQUAL, // >=
  LESS, // <
  LESS_EQUAL, // <=

  // リテラル
  IDENTIFIER, // 識別子
  STRING, // 文字列
  NUBMER, // 数値

  // 予約後
  AND, // and
  CLASS, // class
  ELSE, // else
  FALSE, // false
  FUN, // fun
  FOR, // for
  IF, // if
  NIL, // nil
  OR, // or
  PRINT, // print
  RETURN, // return
  SUPER, // super
  THIS, // this
  TRUE, // true
  VAR, // var
  WHILE, // while,

  EOF // EOF
}

File: php/lox/TokenType.php
<?php

declare(strict_types=1);

namespace Lox;

enum TokenType: string
{
    // Single-character tokens.
    case LEFT_PAREN = 'LEFT_PAREN';
    case RIGHT_PAREN = 'RIGHT_PAREN';
    case LEFT_BRACE = 'LEFT_BRACE';
    case RIGHT_BRACE = 'RIGHT_BRACE';
    case COMMA = 'COMMA';
    case DOT = 'DOT';
    case MINUS = 'MINUS';
    case PLUS = 'PLUS';
    case SEMICOLON = 'SEMICOLON';
    case SLASH = 'SLASH';
    case STAR = 'STAR';

    // One or two character tokens.
    case BANG = 'BANG';
    case BANG_EQUAL = 'BANG_EQUAL';
    case EQUAL = 'EQUAL';
    case EQUAL_EQUAL = 'EQUAL_EQUAL';
    case GREATER = 'GREATER';
    case GREATER_EQUAL = 'GREATER_EQUAL';
    case LESS = 'LESS';
    case LESS_EQUAL = 'LESS_EQUAL';

    // Literals.
    case IDENTIFIER = 'IDENTIFIER';
    case STRING = 'STRING';
    case NUMBER = 'NUMBER';
    case Keywords = 'Keywords';
    case AND = 'AND';
    // case CLASS = 'CLASS'; it's a reserved word in PHP
    case CLAZZ = 'CLAZZ';
    case ELSE = 'ELSE';
    case FALSE = 'FALSE';
    case FUN = 'FUN';
    case FOR = 'FOR';
    case IF = 'IF';
    case NIL = 'NIL';
    case OR = 'OR';
    case PRINT = 'PRINT';
    case RETURN = 'RETURN';
    case SUPER = 'SUPER';
    case THIS = 'THIS';
    case TRUE = 'TRUE';
    case VAR = 'VAR';
    case WHILE = 'WHILE';
    case EOF = 'EOF';
}
